docs: release process steps (#210)
auth0/auth0-cli: CONTRIBUTING.md

We use vendoring, so the last step is required.

## Releasing a new version 

> This is only possible if you're a repository maintainer.

The release is driven by a Github **workflow** triggered when a new **tag** is **created**. The workflow will run the checks and trigger _Goreleaser_ to:
- create the Release with the proper description (changelog)
- upload the binaries for the different architectures
- update https://github.com/auth0/homebrew-auth0-cli with the latest binary reference

To release a new version:

1. pull the latest changes: 
   - `$ git checkout main`
   - `$ git pull origin main`
2. check the latest tag: 
   - `$ git fetch`
   -  `$ git tags`
3. create the **new** tag for the new release. For example, if the latest tag is `v0.1.1`, you should create `v.0.1.1`:
   - `git tag v0.1.1`
4. push the new tag: 
   - `$ git push origin v0.1.2`

> The rest of the process will take place in the github action: https://github.com/auth0/auth0-cli/actions/workflows/goreleaser.yml

Reviewer: ```suggestion
The rest of the process will take place in the github action: https://github.com/auth0/auth0-cli/actions/workflows/goreleaser.yml.
```

> Once the workflow finished, a new release will be available for the create tag.

Reviewer: ```suggestion
Once the workflow finishes, a new release will be available for the newly created tag.
```

We use vendoring, so the last step is required.

## Releasing a new version 

> This is only possible if you're a repository maintainer.

The release is driven by a Github **workflow** triggered when a new **tag** is **created**. The workflow will run the checks and trigger _Goreleaser_ to:
- create the Release with the proper description (changelog)
- upload the binaries for the different architectures
- update https://github.com/auth0/homebrew-auth0-cli with the latest binary reference

To release a new version:

1. pull the latest changes: 
   - `$ git checkout main`
   - `$ git pull origin main`
2. check the latest tag: 
   - `$ git fetch`

>    -  `$git tags`

Reviewer: ```suggestion
   -  `$ git tags`
```

> 3. create the **new** tag for the new release. For example, if the latest tag is `v0.1.1`, you should create `v.0.1.1`:

Reviewer: ```suggestion
3. create the **new** tag for the new release. For example, if the latest tag is `v0.1.1` and you want to release a patch version, you should create `v0.1.2`:
```

>    - `git tag v0.1.1`

Reviewer: ```suggestion
   - `git tag v0.1.2`
```

4. push the new tag: 

>    - `$ git push origin v.0.1.1`

Reviewer: ```suggestion
   - `$ git push origin v.0.1.2`
```

Author: thanks 💤 🤦
